app/styles/ember-rdfa-editor.scss

```diff
@@ -23,6 +22,10 @@
 // OBJECTS
 @import "ember-appuniversum/a-objects";
 
+@import "ember-appuniversum/a-generic";
```

Comment:
Hmm, assuming an app has these imports:

```scss
@import "ember-appuniversum":
@import "ember-rdfa-editor":
```

I _think_ the css will be duplicated if it's imported both here and in `ember-appuniversum.scss`. AFAIK `@import` does a "dumb" copy paste of the code which means some of these files will be present twice in the final output. It's only safe to do for things like variable definitions.

So I think `ember-rdfa-editor` needs to export only it's own custom styles so consuming apps can import it as a "plugin".

The dummy apps simply need to import the Appuniversum styles as well.

Reply:
You're probably right, but for the record, I really, really hate this

Reply:
but the editor references appuniversum styles in its own custom styles, such as variables and utilities
what do I do with those then?

Reply:
I guess I can just assume they'll be there since we peerdep on appuniversum then?

Reply:
Seems so

Reply:
Yea, it's not nice. I think things might improve if we switch to [`@use`](https://sass-lang.com/documentation/at-rules/use) instead of [`@import`](https://sass-lang.com/documentation/at-rules/import). The disclaimer on the `@import` page also seems to suggest this is the way to go. I haven't looked into the specifics myself yet though, but @Dietr might have some insights.

To be fair, the editor already lists ember-appuniversum as a peerDependency which assumes the consumer is using Appuniversum as well, so I don't necessarily think it's a bad thing to expect that the app loads the Appuniversum styles in a way that the editor can depend on that.

Long term (v2 addons) I think it might be better to switch to a "each component imports its own styles" system and discard the Sass setup altogether. I don't think apps configure the Sass variables (or _should_ even) and I assume most of it could be replaced with CSS variables if needed.

Reply:
> but the editor references appuniversum styles in its own custom styles, such as variables and utilities

importing a variables definition file should be safe to do since it doesn't output any css. I'm not sure what you mean with utilities but as long as it doesn't output css by simply importing the file it should be good.